tests/test_declaration.py

```diff
@@ -657,3 +689,53 @@ def test_dict(self):
 
         with self.assertRaises(KeyError):
             del node.style['no-such-property']
+
+    def test_font_shorthand_property(self):
+        node = TestNode(style=CSS())
+        node.layout.dirty = None
+
+        # Check initial value
+        self.assertEqual(node.style.font, 'normal normal normal medium/normal initial')
+
+        # Check Initial values
+        self.assertEqual(node.style.font_style, 'normal')
+        self.assertEqual(node.style.font_weight, 'normal')
+        self.assertEqual(node.style.font_variant, 'normal')
+        self.assertEqual(node.style.font_size, 'medium')
+        self.assertEqual(node.style.line_height, 'normal')
+        self.assertEqual(node.style.font_family, ['initial'])
+
+        # Check individual properties update the unset shorthand
+        node.style.font_style = 'italic'
+        node.style.font_weight = 'bold'
+        node.style.font_variant = 'small-caps'
+        node.style.font_size = '10px'
+        node.style.line_height = '1.5'
+        node.style.font_family = ['"DejaVu Sans"', 'serif']
+        # TODO: Is this the behavior we want?
+        self.assertEqual(node.style.font, 'italic small-caps bold 10.0px/1.5 "DejaVu Sans", serif')
```

Comment:
I'd say this is definitely what `__str__` should return on a font; but the font itself should probably be a wrapper object of some kind.

Reply:
@freakboy3742 would it be ok to just use a dictionary? (as the one we use for the initial font values?)